### ISSUE TITLE
False Positive With --enforce Option?

### ISSUE THREAD

**Issue by DavidMuller** [PyCQA/python-modernize#161]
Thank you for all the hard work on this project!

I was wondering if someone can help me understand why the `--enforce` option is returning a non zero exit code for a file whose only contents is `raise RunTimeError('foo')`:

```bash
$ cat my_file.py 
raise RuntimeError('foo')
$ python-modernize my_file.py --enforce
RefactoringTool: Skipping implicit fixer: idioms
RefactoringTool: Skipping implicit fixer: set_literal
RefactoringTool: Skipping implicit fixer: ws_comma
RefactoringTool: No changes to my_file.py
RefactoringTool: Files that need to be modified:
RefactoringTool: my_file.py
$ echo $?
2 
```

`python-modernize` does not have any suggested changes to the file, but somehow the file still appears in the `Files that need to be modified` section and the exit code is 2?

I'm on `Python 2.7.10` and `modernize 0.6`.

Possibly related SO question:  [`2to3 says "No changes needed", then "files that need to be modified"`](https://stackoverflow.com/questions/40240229/2to3-says-no-changes-needed-then-files-that-need-to-be-modified).

**Comment by takluyver** [comment]
It does sound similar to the SO question, and it's probably a bug. I don't have time to track down where the bug is and whether it's in modernize or in 2to3, but feel free to post here if anyone can pin it down more.

**Comment by larserikh** [comment]
```
$ cat << EOF > hello.py
> 'hello'
> EOF

$ 2to3 hello.py 
RefactoringTool: Skipping optional fixer: buffer
RefactoringTool: Skipping optional fixer: idioms
RefactoringTool: Skipping optional fixer: set_literal
RefactoringTool: Skipping optional fixer: ws_comma
RefactoringTool: No changes to hello.py
RefactoringTool: Files that need to be modified:
RefactoringTool: hello.py

$ 2to3 -x unicode hello.py 
RefactoringTool: Skipping optional fixer: buffer
RefactoringTool: Skipping optional fixer: idioms
RefactoringTool: Skipping optional fixer: set_literal
RefactoringTool: Skipping optional fixer: ws_comma
RefactoringTool: No files need to be modified.
```
It seems to be the unicode fixer in lib2to3 that's causing these false positives.

**Comment by rouge8** [comment]
It seems like there's several fixers with this issue, including numliterals, ws_comma, and dict_six. 😕  (Tested by running `python-modernize -f $fix` with each individual fix over a file with false positives)

**Comment by cmin764** [comment]
Any news here? Is pretty bad that we can't use that as a pre-commit hook since is returning these false-positives.

**Comment by AWhetter** [comment]
It seems like this issue comes from lib2to3. I have a pull request in with cpython to fix those issues. With those fixes in place I'm not seeing any modernize code that's causing this issue from my testing.

**Comment by graingert** [comment]
@AWhetter lib2to3 is deprecated, can you make your PR into fissix?

> It seems like this issue comes from lib2to3. I have a pull request in with cpython to fix those issues.

**Comment by graingert** [comment]
@DavidMuller looks like this is fixed in `pip install -U fissix "modernize>=0.8rc0"`
